### ISSUE TITLE
Type guards doesn't work on multilevel if statements

### ISSUE THREAD

**Issue by tinganho** [Microsoft/TypeScript#5100]
I just noticed that type guards doesn't work on multilievel if statements? Is it by design?
```typescript
var t: string | boolean | number;
if (typeof t === 'boolean' || typeof t === 'number') {
	if (typeof t === 'number') {		
	}
	else {
		t // should be boolean, but is string | boolean
	}
}
```

**Comment by DanielRosenwasser** [comment]
I wonder if this is at all related to #4874, but it definitely seems like a bug.

**Comment by tinganho** [comment]
@DanielRosenwasser I think it is related.

**Comment by weswigham** [comment]
Fixed by #5442.
